typed.go

```diff
@@ -41,15 +44,55 @@ encoding definition for standard StarkNet Domain messages
 func (dm Domain) FmtDefinitionEncoding(field string) (fmtEnc []*big.Int) {
 	switch field {
 	case "name":
-		fmtEnc = append(fmtEnc, types.StrToFelt(dm.Name).Big())
+		nameFelt := strToFelt(dm.Name)
+		nameBig, ok := feltToBig(nameFelt)
+		if ok {
+			fmtEnc = append(fmtEnc, nameBig)
+		}
 	case "version":
-		fmtEnc = append(fmtEnc, types.StrToFelt(dm.Version).Big())
+		versionFelt := strToFelt(dm.Version)
+		versionBig, ok := feltToBig(versionFelt)
+		if ok {
+			fmtEnc = append(fmtEnc, versionBig)
+		}
 	case "chainId":
-		fmtEnc = append(fmtEnc, types.StrToFelt(dm.ChainId).Big())
+		chainIdFelt := strToFelt(dm.ChainId)
+		chainIdBig, ok := feltToBig(chainIdFelt)
+		if ok {
+			fmtEnc = append(fmtEnc, chainIdBig)
+		}
```

Comment:
could we make a small method for this logic as it repeats multiple times in this switch?